style_guides/styling.md

```diff
@@ -0,0 +1,84 @@
+# Styling Grafana
+
+## Emotion
+
+[Emotion](https://emotion.sh/docs/introduction) is our default-to-be approach to styling React components. It provides a way for styles to be a consequence of properties and state of a component.
+
+### Usage
+
+#### Basic styling
+
+For styling components use Emotion's `css` function
+
+```tsx
+import { css }  from 'emotion';
+
+
+const ComponentA = () => {
+  return (
+    <div className={css`background: red;`}>
+      As red as you can ge
+    </div>
+  );
+}
+```
+
+#### Styling complex components
+
+In more complex cases, especially when you need to style multiple DOM elements in one component or when your styles that depend on properties and/or state, you should create a helper function that returns an object with desired stylesheet. Let's say you need to style a component that has different background depending on the theme:
+
+```tsx
+import { css, cx }  from 'emotion';
+import { GrafanaTheme, useTheme, selectThemeVariant } from '@grafana/ui';
+
+const getStyles = (theme: GrafanaTheme) => {
```

Comment:
Is this not missing the wrapper that adds the memoize-one call? wonder if we can re-export css & cx from grafana@/ui  and then a styleFactory((theme) =>

Reply:
Yeah I should do that, sorry did not get to it yet. Not sure if needs to wait for me (will probably get to it tomorrow) and I can just update it in my PR.

Reply:
I mean to add that but meetings sucked me in in the afternoon:) @aocenas no need to worry about that, see 625c2a361cde68d9de2e5d70772eb3546d351e41 for implementation

Reply:
@torkelo good idea with re-exporting css/cx via grafana-ui